### DISCUSSION TITLE
Reupload previous batch with different file content

### DISCUSSION THREAD

**Discussion by sadrzadehsina** [rpldy/react-uploady#761] [Q&A]
I'm implementing a crop function that allows users to crop the same image multiple times. The issue is that while the uploader successfully uploads the cropped image on the first attempt, subsequent attempts don't trigger the uploader to upload.

Is there a way to clear previous uploads when they’re in the finished state, or perhaps to create a new batch using a constructor?

Just to note, I created the preview component entirely from scratch and didn't use the `UploadPreview` component.

**Comment by yoavniran** [comment]
Hey @sadrzadehsina 
it sounds like a cool use-case, wish I had time to create an example for it... 😓  

From what you describe, I think the best way would be to create new uploads using the Context's [upload](https://react-uploady.org/docs/api/context/#upload) method.
These would be treated as completely new items and will have their own upload cycle.

Let me know how it goes.

**Comment by sadrzadehsina** [comment]
Thanks @yoavniran for elaborating on this. I had a similar idea, but things aren't working as expected. Here’s what I’m dealing with:

![SCR-20241103-jccw](https://github.com/user-attachments/assets/e08843e2-47a7-402f-ad9c-c22efea6701f)

I need a crop function alongside the uploader, and I want the cropped selection to trigger a re-upload. This way, the cropped image stays synchronized with what the user sees.

The issue is that whenever I use the upload function each time the cropped image changes, it resets everything, causing me to lose the previous crop selection.

I am trying to create a demo for my problem so I can share it with you.

**Reply by yoavniran** [reply]
What you described sounds like a [sandbox](https://codesandbox.io/p/sandbox/react-uploady-crop-and-upload-uyl4e) I created a while ago:

<img width="927" alt="image" src="https://github.com/user-attachments/assets/4ef6ca54-5863-4640-84e7-663ae1b27c52">

It allows you to crop the image and upload only the cropped area.

It replaces the bytes to upload so keeps a single upload as far as Uploady is concerned, leveraging [withRequestPreSendUpdate](https://react-uploady.org/docs/api/hocs/withRequestPreSendUpdate/) HOC.

Hopefully this can help you get started.
